cmake/macros/macro_deal_ii_add_test.cmake

```diff
@@ -460,6 +475,35 @@ function(deal_ii_add_test _category _test_name _comparison_file)
           RUNTIME_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${_target_short}"
           )
 
+        add_dependencies(compile_test_executables ${_target})
+      endif()
+
+      #
+      # Add a top level target to compile the test:
+      #
+
+      if(_shared_target AND NOT TARGET ${_test_executable_target})
+        add_custom_target(${_test_executable_target}
+          COMMAND echo "${_test_executable_full}: BUILD successful."
+          COMMAND echo "${_test_executable_full}: RUN skipped."
+          COMMAND echo "${_test_executable_full}: DIFF skipped."
+          COMMAND echo "${_test_executable_full}: PASSED."
+          DEPENDS ${_target}
+          )
+        add_test(NAME ${_test_executable_full}
+          COMMAND ${CMAKE_COMMAND}
+            -DTRGT=${_test_executable_target}
+            -DTEST=${_test_executable_full}
+            -DEXPECT=PASSED
+            -DBINARY_DIR=${CMAKE_BINARY_DIR}
+            -P ${DEAL_II_PATH}/${DEAL_II_SHARE_RELDIR}/scripts/run_test.cmake
+          WORKING_DIRECTORY ${_test_directory}
+          )
+        set_tests_properties(${_test_executable_full} PROPERTIES
+          LABEL "test_dependency"
+          TIMEOUT ${TEST_TIME_LIMIT}
+          FIXTURES_SETUP ${_test_executable_full}
```

Comment:
Is this right? This suggests that the test is its own setup?

Reply:
This is correct. A "FIXTURE" is an arbitrary name for declaring ordering requirements. The important bit is to `FIXTURE_SETUP` and to `FIXTURE_REQUIRES` the same name.